### DISCUSSION TITLE
OpenAI Whisper glitches - repeated lines and timing off

### DISCUSSION THREAD

**Discussion by Windrider6** [SubtitleEdit/subtitleedit#7183] [Q&A] [answered]
I'm using OpenAI Whisper in Subtitle Edit, and it is glitching. Lines will be repeated several times, sometimes a few lines repeated. This will throw the timing out, and lines of dialog will be missed. This is also happening with Faster Whisper.

Anyone else see this? Is there a solution to the problem?

**Comment by niksedk** [comment]
Hopefully this will improve in a future version of Whisper.

You can try to experiment with some of the parameters, like "--beam_size" (OpenAI) or "--max-context" (CPP/Const-me).

A parameter (the Advanced button) can be added when using latest SE beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.13/SubtitleEditBeta.zip

![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/61bf02db-81ed-4ce0-b5e4-8e2a0dda0914)

**Reply by Windrider6** [reply]
OK.  Questions:
1. What is the default beam size value and the range of values?
2. What would you recommend the beam size be for a Windows 10 computer, 3.7 GHz processor and 32 GB of RAM?

**Reply by niksedk** [reply]
I'm not sure what the range is, probably around -1 to 15... try with something like 1-3.
` --beam_size BEAM_SIZE         number of beams in beam search, only applicable when temperature is zero (default: 5)`

Here is another parameter you can try:
`--best_of BEST_OF                   number of candidates when sampling with non-zero temperature (default: 5)`

**Reply by Windrider6** [reply]
Thank you.  I read "Introduction to Beam Search Algorithm", but cannot understand it.
What does changing the beam size actually do in practical matters like processing power, RAM, and processing time, when the number is lower or higher than the default?
Ug.  What the heck is "temperature" supposed to be?  It's above freezing in here ...

**Reply by Windrider6** [reply]
I've tried several options now, and all of them have produced the repeating lines glitch:
CTranslate2 (Faster Whisper): medium (1.5 GB)
--beam_size 1
CTranslate2 (Faster Whisper): large (2.9 GB):
--beam_size 1
--beam_size 15
--temperature 0 --beam_size 10 --patience 2
--temperature 0.2 --best_of 10 --patience 2

I clipped out a 1 minute section of the TV show where the last test had the glitch, then ran the process with the same parameters, and did not get the glitch.

I don't know what to do now.

**Comment by Purfview** [comment, marked as answer]
Is it stuck in repeating lines or just repeats some lines and then continues normally?
Try [Standalone Faster Whisper](https://github.com/Purfview/whisper-standalone-win).

Would be helpful if you could share an audio sample to reproduce the issue.

**Reply by Windrider6** [reply]
It repeats lines, then continues normally.  The timing is put off.  Faults in the subtitles occur before, during, and after the glitch.

I clipped out a 1 minute section of the TV show where the last test had the glitch, then ran the process with the same parameters, and did not get the glitch.
CTranslate2 (Faster Whisper): large (2.9 GB):
   --temperature 0.2 --best_of 10 --patience 2

The entire video that I have been testing is "Dead Man's Gun S01E18 The Resurrection of Joe Wheeler.mkv", which is 44 min 29 s, 232 MB.  I don't know if anyone cares that I share it somehow.

I don't know what to do now.

Is it possible to use Standalone Faster Whisper to produce subtitles that are timed properly?

**Reply by Purfview** [reply]
> Is it possible to use Standalone Faster Whisper to produce subtitles that are timed properly?

Yes. Check if you get this glitch with it.

**Reply by Windrider6** [reply]
I give up.  I don't understand how to use Python.  I tried following the instructions at [https://github.com/guillaumekln/faster-whisper](url), but it will not work in python console or as a py file.

I really want to create subtitles for some TV shows that do not have any available, but the number of problems I am encountering are just too much for me.

**Reply by Purfview** [reply]
Why you are going to another repo if I posted you this link -> https://github.com/Purfview/whisper-standalone-win
In releases you will find ready to run exe. you don't need Python at all.

PS:
You can run it in console or in Subtitle Edit, Btw, latest Subtitle Edit beta has option to download it automatically.

**Reply by Windrider6** [reply]
I misunderstood.

Used Standalone Faster Whisper and did not get those repeated lines glitch.  Great!  Why is that?  How is it different?

Fixing the word errors and timing still takes me a long time.

**Reply by Purfview** [reply]
Because OpenAI's pseudo VAD is not good.
Of course you need to spend some time in fixing subtitles, model is not 100% accurate.
